lib/fizzy/execute.cpp

```diff
@@ -625,23 +622,24 @@ std::unique_ptr<Instance> instantiate(Module module,
     return instance;
 }
 
-execution_result execute(
-    Instance& instance, FuncIdx func_idx, std::vector<uint64_t> args, int depth)
+execution_result execute(Instance& instance, FuncIdx func_idx, span<const uint64_t> args, int depth)
 {
     assert(depth >= 0);
     if (depth > CallStackLimit)
         return {true, {}};
 
     if (func_idx < instance.imported_functions.size())
-        return instance.imported_functions[func_idx].function(instance, std::move(args), depth);
+        return instance.imported_functions[func_idx].function(instance, args, depth);
 
     const auto code_idx = func_idx - instance.imported_functions.size();
     assert(code_idx < instance.module.codesec.size());
 
     const auto& code = instance.module.codesec[code_idx];
     auto* const memory = instance.memory.get();
 
-    std::vector<uint64_t> locals = std::move(args);
+    std::vector<uint64_t> locals;
+    locals.reserve(args.size() + code.local_count);
+    std::copy_n(args.begin(), args.size(), std::back_inserter(locals));
     locals.resize(locals.size() + code.local_count);
```

Comment:
Do we need `reserve` + `resize` or can we stay with one?

Reply:
Also this seems to now expand it to `args.size() + local_count + local_count`.

Reply:
We could first `resize` here, then just `copy_n` without `back_inserter`

Reply:
This is "minimal work" version.
1. Firstly we allocate memory for both args and locals (`locals.size()` is still 0 at this point).
2. We copy args.
3. We fill locals with zero.

Reply:
In the end I used the what @gumb0 suggested. Micro-benchmarks included.